prow/git/v2/interactor.go

```diff
@@ -461,6 +461,19 @@ func (i *interactor) FetchCommits(commitSHAs []string) error {
 
 // RemoteUpdate fetches all updates from the remote.
 func (i *interactor) RemoteUpdate() error {
+	// We might need to refresh the token for accessing remotes in case of GitHub App auth (ghs tokens are only valid for
+	// 1 hour, see https://github.com/kubernetes/test-infra/issues/31182).
+	// Therefore, we resolve the remote again and update the clone's remote URL with a fresh token.
+	remote, err := i.remote()
+	if err != nil {
+		return fmt.Errorf("could not resolve remote for updating: %w", err)
+	}
+
+	i.logger.Info("Setting remote URL")
+	if out, err := i.executor.Run("remote", "set-url", "origin", remote); err != nil {
```

Comment:
We usually try not to mutate remotes and run against URLs wherever possible since that makes it much easier to run this code on top of an existing cached `git` repo of any shape - are we guaranteed there's a remote called `origin` and that's something we want to / can change here?

Reply:
Yeah, I looked into this but the `git remote update` command only takes a remote name arg but not a remote URL. From the docs, it is explicitly supposed to be rung against repos cloned with `--mirror` to update all refs from the default remote, which is what we do in `MirrorClone`. The default remote is named `origin` by default, we don't change it anywhere via `clone.defaultRemoteName` or the `clone -o` param.

I only see one usage of `RemoteUpdate` in `maybeCloneAndUpdatePrimary`: https://github.com/kubernetes/test-infra/blob/d635123c32b5134706e5c9aed7d11ea0fa7188e2/prow/git/v2/client_factory.go#L478
Here, we are guaranteed that there is a remote called `origin` as we cloned the primary using `MirrorClone`.
As `RemoteUpdate` is only a method on the internal `cacher` interface, I would say this change is safe.

Reply:
Awesome, thanks for doing that analysis. Let's merge this in!